[DOC] Update and expand profiling types doc (#3659)
grafana/pyroscope: docs/sources/configure-client/trace-span-profiles/_index.md

Reviewer: I wonder if we want to explain a common friction point we have with a lot of customers who use span profiles. That is: Span profiling is not effective on short spans (<20ms) as there are only a couple sampled profiles. This results in a statistically inaccurate profile.

Ideally we should only trust span profiles that are maybe 100ms or longer. (Not sure on the specific threshold, perhaps the rest of the team can weigh in here)

Author: I've added an admonition noting that span profiling is not effective on short spans, using your wording. I didn't put a comment about the specific size to trust. We can always update this information.

Author: Ryan weighed in with a suggested edit, which I've added.

- Seamless integration: Smoothly transition from a high-level trace overview to detailed profiling of specific trace spans within Grafana’s trace view
- Efficiency and cost savings: Quickly identify and address performance issues, reducing troubleshooting time and operational costs

{{< admonition type="note">}}

> Span profiling is not effective on short spans (<20ms) because there are only a couple sampled profiles.
> This results in a statistically inaccurate profile.

Author: ```suggestion
Span profiling is only effective on spans longer than 20ms to ensure statistical accuracy. 
```
